File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    answer = True
    st = deque()

    for c in s:
        if(c == '('):
            st.append(c)
        else:
            if not st:
                return False
            else:
                st.pop()
    
    if st:
        return False
    else:
        return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>
using namespace std;

bool solution(string s)
{
    stack<char> st;
    int idx = 0;
    
    while(idx < s.size())
    {
        if(s[idx] == '(') 
            st.push('(');
        else
        {
            if(st.empty()) return false;
            st.pop();
        }
        idx++;
    }
    return st.empty();
}